### ISSUE TITLE
Inherited template typedef does not extend template instantiation

### ISSUE THREAD

**Issue by scott-zhong** [doxygen/doxygen#10760]
**Describe the bug**
Inherited template `typedef` does not extend template instantiation.  The problem started with 1.8.18 and continues with Doxygen trunk (cbb48edb43715e3420a69d69c21e6e3c4f998894).

```
/**
 * Base docs
 */
template <class Type>
class Base
{
public:
    /**
     * A typedef
     */
    typedef typename Type::element_type element_type;
};

/**
 * MyCharType docs
 */
class MyCharType
{
public:
    /**
     * A typedef
     */
    typedef char element_type;
};

/**
 * Derived docs
 */
class Derived : public Base<MyCharType>
{
};
```

**Screenshots**
Doxygen 1.8.17:
![image](https://github.com/doxygen/doxygen/assets/32584126/38c30b9c-4c1a-43d6-9d21-fb0f78a26989)

Doxygen 1.8.18 - Doxygen trunk (cbb48edb43715e3420a69d69c21e6e3c4f998894):
![image](https://github.com/doxygen/doxygen/assets/32584126/b9db89a0-8bb5-40fb-87d0-b0b496ddf564)

**To Reproduce**
[testcase.zip](https://github.com/doxygen/doxygen/files/14672474/testcase.zip)

**Expected behavior**
`Type` should be expanded to `MyCharType`.

**Comment by doxygen** [comment]
@scott-zhong Please verify if the referenced commit fixes the problem for you. Do not close the issue, this will be done automatically when the next official release becomes available.

**Comment by scott-zhong** [comment]
@doxygen I have verified this fixes the problem.  Thank you!!!

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.11.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
